tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/Configuration/HandsOffConfigurationTests.cs

```diff
@@ -0,0 +1,199 @@
+            {
+                localDestPathFolder = "C:\\ProgramData\\Datadog\\";
+                fleetDestPathFolder = "C:\\ProgramData\\Datadog\\managed\\datadog-agent\\stable\\";
+                localDestPath = Path.Combine(localDestPathFolder, applicationMonitoringYaml);
+                fleetDestPath = Path.Combine(fleetDestPathFolder, applicationMonitoringYaml);
+            }
+            else if (EnvironmentTools.IsOsx())
+            {
+                // even if it's skipped on mac keep it for local debugging purpose
+                localDestPathFolder = "/opt/datadog-agent/etc/";
+                fleetDestPathFolder = "/opt/datadog-agent/etc/stable/";
+                localDestPath = Path.Combine(localDestPathFolder, applicationMonitoringYaml);
+                fleetDestPath = Path.Combine(fleetDestPathFolder, applicationMonitoringYaml);
+            }
+
+            if (File.Exists(localDestPath))
+            {
+                File.Delete(localDestPath);
+            }
+
+            if (File.Exists(fleetDestPath))
+            {
+                File.Delete(fleetDestPath);
+            }
+
+            var localLines = localValues.Select(localValue => $"  {localValue.Key}: {localValue.Value}").ToList();
+            if (localLines.Any())
+            {
+                if (!Directory.Exists(localDestPathFolder))
+                {
+                    Directory.CreateDirectory(localDestPathFolder);
+                }
+
+                localLines.Insert(0, "apm_configuration_default:");
+                File.AppendAllLines(localDestPath, localLines);
+            }
+
+            var fleetLines = fleetValues.Select(fleetValue => $"  {fleetValue.Key}: {fleetValue.Value}").ToList();
+            if (fleetLines.Any())
+            {
+                if (!Directory.Exists(fleetDestPathFolder))
+                {
+                    Directory.CreateDirectory(fleetDestPathFolder);
+                }
+
+                fleetLines.Insert(0, "apm_configuration_default:");
+                File.AppendAllLines(fleetDestPath, fleetLines);
+            }
+
+            return (localDestPath, fleetDestPath);
+        }
+
+        private async Task SetupAndCheck(string logFolder, string? localKey, string? localValue, string? fleetKey, string? fleetValue, string logKey, string expectedValue)
+        {
+            // SkipOn.Platform(SkipOn.PlatformValue.MacOs);
+
+            var logDir = Path.Combine(LogDirectory, logFolder);
+            Directory.CreateDirectory(logDir);
+            SetEnvironmentVariable(ConfigurationKeys.LogDirectory, logDir);
+
+            using var agent = MockTracerAgent.Create(Output, useTelemetry: false);
+            Output.WriteLine($"Assigned port {agent.Port} for the agentPort.");
+
+            var processName = EnvironmentHelper.IsCoreClr() ? "dotnet" : "Samples.Console";
+            using var logEntryWatcher = new LogEntryWatcher($"{LogFileNamePrefix}{processName}*", logDir, Output);
```

Comment:
I would suggest not bothering with LogEntryWatcher tbh. That's useful when you need to read the logs in realtime, but it's prone to race conditions. Just read the logs after the sample has exited instead